libcodechecker/analyze/log_parser.py

```diff
@@ -47,75 +125,50 @@ def get_compiler_includes(compiler, lang, compile_opts, extra_opts=None):
-                do_append = False
-            if do_append:
-                # On OSX there are framework includes,
-                # where we need to strip the "(framework directory)" string.
-                # For instance:
-                # /System/Library/Frameworks (framework directory)
-                fpos = line.find("(framework directory)")
-                if fpos == -1:
-                    include_paths.append("-isystem " + line)
-                else:
-                    include_paths.append("-isystem " + line[0:fpos-1])
-
-            if line.startswith(start_mark):
-                do_append = True
-
-    except OSError as oerr:
-        LOG.error("Cannot find include paths: " + oerr.strerror + "\n")
-    return include_paths
+    err = get_compiler_err(cmd)
+    if output_path is not None:
+        dump_compiler_info(output_path, compiler_includes_dump_file,
+                           {"compiler": compiler, "includes": err})
+    return parse_compiler_includes(err)
 
 
-def get_compiler_target(compiler):
+def get_compiler_target(compiler, output_path=None):
     """
     Returns the target triple of the given compiler as a string.
-
     """
-    target_label = "Target:"
-    target = ""
 
     cmd = compiler + ' -v'
     LOG.debug("Retrieving target platform information via '" + cmd + "'")
 
-    try:
-        proc = subprocess.Popen(shlex.split(cmd),
-                                stdin=subprocess.PIPE,
-                                stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-
-        _, err = proc.communicate("")
-        for line in err.splitlines(True):
-            line = line.strip().split()
-            if line[0] == target_label:
-                target = line[1]
+    err = get_compiler_err(cmd)
+    if output_path is not None:
+        dump_compiler_info(output_path, compiler_target_dump_file,
+                           {"compiler": compiler, "target": err})
+    return parse_compiler_target(err)
 
-    except OSError as oerr:
-        LOG.error("Cannot find compiler target: " + oerr.strerror + "\n")
 
-    return target
+def remove_file_if_exists(filename):
+    try:
+        os.remove(filename)  # Throws if file does not exsist.
+    except OSError as e:
+        if e.errno != errno.ENOENT:  # errno.ENOENT = no such file or directory
+            raise  # Re-raise exception if a different error occurred.
 
 
-def parse_compile_commands_json(logfile, add_compiler_defaults=False):
+def parse_compile_commands_json(logfile, output_path=None,
+                                add_compiler_defaults=False):
     import json
     # The add-compiler-defaults is a deprecated argument
     # and we always perform target and include auto-detection.
     add_compiler_defaults = True
     LOG.debug('parse_compile_commands_json: ' + str(add_compiler_defaults))
 
+    if output_path is not None:
+        remove_file_if_exists(os.path.join(output_path,
```

Comment:
Is it always applicable for incremental analysis to remove these files? What if analysis configuration changes?

Reply:
It is not a problem. If the analysis configuration changes the compile database still remains the same. Compiler includes and targets do not depend on analysis configuration.